Automattic/simplenote-macos
# Decoupling VersionsViewController

### Details:
- Splits the `VersionsViewController` UI from the MainNib: new specific nib
- Encapsulates all of its properties only in `VersionsViewController`: NoteEditorViewController will only access them thru the VersionsViewController Public API
- Swiftens VersionsVC

Our goal is to have the Main NIB as clean as possible, in order to implement NSSplitViewController. And each one of the current ViewControllers should not be coupled with each other.

@aerych May I trouble you with this one sir?
Thaaank you!!

Ref. #458

### Test
1. Open any of your notes
2. Click over the `Versions` button in the editor (4th from the right!)

- [x] Verify the Restore action is disabled for the "newest" version
- [x] Verify that moving the slider results in an older revision onscreen
- [x] Verify that clicking the Restore button saves the document (try relaunching + verify it sticks!)
- [x] Verify the Restore Popover reacts well to Light / Dark themes

### Release
These changes do not require release notes.
